langserver/general.go

```diff
@@ -58,16 +58,16 @@ func (s *server) Initialize(_ context.Context, _ *protocol.ParamInitialize) (*pr
 
 // Initialized receives a confirmation by the client that the connection has been initialized
 // required by the protocol.Server interface.
-func (s *server) Initialized(ctx context.Context, _ *protocol.InitializedParams) (err error) {
+func (s *server) Initialized(ctx context.Context, _ *protocol.InitializedParams) error {
 	s.stateMu.Lock()
 	defer s.stateMu.Unlock()
 
 	if s.state != serverInitializing {
 		return errors.New("cannot initialize server: wrong server state")
 	}
 
-	if s.config.PrometheusURL != "" {
-		if err := s.connectPrometheus(s.config.PrometheusURL); err != nil {
+	if len(s.prometheusURL) > 0 {
```

Comment:
Just out of curiosity:

It there a specific reason why you prefer `len(str) > 0 ` over `str != ""`?

Reply:
just to avoid the nasty issue when you would like to compare an empty string but you put a space inside and so it's not anymore an empty string